apps/steercode-app-electron/src/renderer/src/lib/models/types/user.type.ts

```diff
@@ -0,0 +1,22 @@
+import type { RepositoryOption } from './conversation.type';
+
+export type StripeInfo = {
+  customer_id: string;
+  subscription_id: string | null;
+  subscribed_until: string | null;
+};
+
+export type User = {
+  id: string;
+  projects: RepositoryOption[];
+  stripe: StripeInfo | null;
+  isPremium: () => boolean;
+};
+
+export function isPremium(user: User): boolean {
+  return (
+    user.stripe !== null &&
+    user.stripe.subscribed_until !== null &&
+    new Date() < new Date(user.stripe.subscribed_until)
```

Comment:
use `Date.now() < new Date(...).getTime().` Comparing Date objects does not work correctly.

Reply:
move this function to user store please. keep the `*.type.ts` files clean so they include only type definitions.